python/pyiceberg/expressions/visitors.py

```diff
@@ -526,7 +528,7 @@ def visit_equal(self, term: BoundTerm, literal: Literal[Any]) -> bool:
         if lower > literal.value:
             return ROWS_CANNOT_MATCH
 
-        upper = _from_byte_buffer(term.ref().field.field_type, field.lower_bound)
+        upper = _from_byte_buffer(term.ref().field.field_type, field.upper_bound)
```

Comment:
@Fokko, can you check why tests didn't catch this case? We should ensure this is tested properly and separate the fix into its own PR.